packages/tempus-core-services/src/services/TempusPoolService.ts

```
@@ -304,4 +335,16 @@ export class TempusPoolService {
     }
     throw new Error(`TempusPoolService - getFeesConfig() - Address '${address}' is not valid`);
   }
+
+  private async getTokenPrecision(tokenAddress: string): Promise<number | null> {
+    if (!this.chain) {
+      return null;
+    }
+
+    const contract = new ERC20Contract(this.chain, tokenAddress);
+    return contract
+      .decimals()
+      .then(decimals => decimals)
+      .catch(() => DEFAULT_DECIMAL_PRECISION);
```

It's not a good idea to silently return invalid token precision in case call fails - this will just mask the error and cause issues in the app because we will end up using wrong token precision.